src/datasets/formatting/formatting.py

```diff
@@ -303,7 +319,9 @@ def key_to_query_type(key: Union[int, slice, range, str, Iterable]) -> str:
 
 
 def query_table(
-    pa_table: pa.Table, key: Union[int, slice, range, str, Iterable], indices: Optional[pa.lib.UInt64Array] = None
+    pa_table: Union[pa.Table, "Table"],
```

Comment:
since you're not inside the `Table` class body you could refer to it directly rather than with a string?

Reply:
also unclear why you'd accept both and not just your new abstraction?